src/Catalog/ValidatePackageHashHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    /// <summary>
+    /// Validates that packages in the packages container have the correct MD5 Content hash on their blob's properties.
+    /// </summary>
+    public class ValidatePackageHashHandler : IPackagesContainerHandler
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ITelemetryService _telemetryService;
+        private readonly ILogger<ValidatePackageHashHandler> _logger;
+
+        public ValidatePackageHashHandler(
+            HttpClient httpClient,
+            ITelemetryService telemetryService,
+            ILogger<ValidatePackageHashHandler> logger)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task ProcessPackageAsync(CatalogIndexEntry packageEntry, ICloudBlockBlob blob)
+        {
+            await blob.FetchAttributesAsync(CancellationToken.None);
+
+            if (blob.ContentMD5 == null)
```

Comment:
Can `ContentMD5` be an empty string? Wrong length? Invalid content?
Are we OK with it if it can?

Reply:
That's covered by line 59, and will be tracked by `TrackPackageHasIncorrectHash`